### PR TITLE
Add specs for EmailTemplatesController

### PR DESCRIPTION
**Pull request by gschlager**
@tgxworld Here are the tests for https://github.com/discourse/discourse/commit/1a378126259b1461bb935d1385778842d33b088a
(And yes, I found and corrected a bug while writing those tests 😉 )

First time using `shared_examples`... I hope I'm using them correctly.

### CODE REVIEW

**Comment by tgxworld** [comment]
@gschlager Thank you! Could we write the controller specs as request specs instead? Controller specs are no longer recommended by the Rails/Rspec team in Rails 5 and beyond.

File: spec/controllers/admin/email_templates_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    expect(Admin::EmailTemplatesController < Admin::AdminController).to eq(true)
+  end
+
+  context "while logged in as an admin" do
+    before do
+      @user = log_in(:admin)
+    end
+
+    after do
+      UserHistory.delete_all
```
Comment (tgxworld) [inline spec/controllers/admin/email_templates_controller_spec.rb]: Hmm do we need this here? The transaction should be rollbacked after each test.

File: spec/controllers/admin/email_templates_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,215 @@
+require 'rails_helper'
+
+describe Admin::EmailTemplatesController do
+
+  it "is a subclass of AdminController" do
```
Comment (tgxworld) [inline spec/controllers/admin/email_templates_controller_spec.rb]: I know we use this in a bunch of places in our code base but I'm personally not a fan of it. I would rather we write a simple spec for an anon user to verify that the access is invalid.

File: spec/controllers/admin/email_templates_controller_spec.rb (hunk 1 above)
Comment (gschlager) [reply spec/controllers/admin/email_templates_controller_spec.rb]: Not really. It just was the easiest way to solve a particular problem in a test. 😉 I'm going to change it.

**Comment by gschlager** [comment]
@tgxworld I'm now using Request specs. I had to set an accept header for PUT and DELETE since `.json` in the URL doesn't work in this case. I also added tests for (anonymous) users instead of using the `is a subclass of AdminController` spec.

**Comment by tgxworld** [comment]
@gschlager Looks good to me :+1:

**Pull request merged**